tests/test_packages/test_skills/test_echo.py

```diff
@@ -20,134 +20,54 @@
 """This test module contains the integration test for the echo skill."""
 
 import os
-import shutil
 import signal
-import subprocess  # nosec
-import sys
-import tempfile
 import time
-from pathlib import Path
 
 import pytest
 
-import yaml
+from aea.test_tools.test_cases import AEATestCase
 
-from aea.cli import cli
-from aea.configurations.base import AgentConfig, DEFAULT_AEA_CONFIG_FILE, PublicId
-from aea.mail.base import Envelope
-from aea.protocols.default.message import DefaultMessage
-from aea.protocols.default.serialization import DefaultSerializer
 
```

Comment:
You could create the derived class here with the methods from line 182-272 and then use it below.